lighthouse-core/gather/gather-runner.js

```diff
@@ -430,13 +430,16 @@ class GatherRunner {
     }
   }
 
+  /**
+   * @return {!ComputedArtifacts}
+   */
   static instantiateComputedArtifacts() {
     const computedArtifacts = {};
     require('fs').readdirSync(path.join(__dirname, 'computed')).forEach(function(file) {
       // Drop `.js` suffix to keep browserify import happy.
       file = file.replace(/\.js$/, '');
       const ArtifactClass = require('./computed/' + file);
-      const artifact = new ArtifactClass();
+      const artifact = new ArtifactClass(computedArtifacts);
```

Comment:
> really not a fan of this intertwined computedArtifacts, artifact object, but I guess it works for now

sure, it uses inheritance for delegation, but on the plus side it guarantees invariants on our artifact dependency graph and it's nicely isolated, so it's ready for an easy refactor at any time :)